File: src/main/java/ac/kr/deu/connect/luck/event/application/EventApplicationRepository.java
package ac.kr.deu.connect.luck.event.application;

import ac.kr.deu.connect.luck.event.entity.Event;
import ac.kr.deu.connect.luck.user.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface EventApplicationRepository extends JpaRepository<EventApplication, Long> {

    /**
     * 이벤트 신청자 목록 조회 (행사 관리자용 API)
     *
     * @param event 조회 하려는 이벤트
     * @return 이벤트에 신청한 신청자 목록
     */
    List<EventApplication> findAllByEvent(Event event);

    List<EventApplication> findAllByEventId(Long eventId);

    /**
     * 이벤트 신청자 목록 조회 (푸드트럭 매니저용 API)
     *
     * @param user 조회 하려는 푸드트럭 매니저
     * @return 푸드트럭 매니저가 신청한 이벤트 목록
     */
    List<EventApplication> findAllByFoodTruckManager(User user);


    List<EventApplication> findAllByEventManager(User user);

    List<EventApplication> findAllByEventManagerAndEventId(User user, Long eventId);
}

File: src/main/kotlin/ac/kr/deu/connect/luck/event/application/repository/EventApplicationRepository.kt
package ac.kr.deu.connect.luck.event.application.repository

import ac.kr.deu.connect.luck.event.application.entity.EventApplication
import ac.kr.deu.connect.luck.event.application.entity.EventApplication.ApplicationId
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface EventApplicationRepository : JpaRepository<EventApplication, ApplicationId> {
    /**
     * 행사별 신청 목록 조회 메소드
     * @param eventId 행사 ID
     * @param pageable 페이징 정보
     * @return 행사별 신청 목록
     */
    fun findByIdEventId(
        eventId: Long,
        pageable: Pageable,
    ): Page<EventApplication>

    /**
     * 푸드트럭별 신청 목록 조회 메소드
     * @param foodTruckId 푸드트럭 ID
     * @param pageable 페이징 정보
     * @return 푸드트럭별 신청 목록
     */
    fun findByIdFoodTruckId(
        foodTruckId: Long,
        pageable: Pageable,
    ): Page<EventApplication>
}
